src/dns/rfc1035.cc

```diff
@@ -252,50 +248,46 @@ rfc1035NameUnpack(const char *buf, size_t sz, unsigned int *off, unsigned short
     assert(ns > 0);
     do {
         if ((*off) >= sz) {
-            RFC1035_UNPACK_DEBUG;
+            debugs(DEBUG_SECTION, DBG_DATA, "unpack error: parsing beyond end");
```

Comment:
I see several problems in f12702b8d0dd24f93c5b5b3c2db1469e6e59b9ae. @kinkie, do you want me to start detailing them in this PR? I recommend undoing all src/dns/rfc1035.cc (and related) modifications instead, to avoid arguing about this out-of-scope change in this PR.

Reply:
> If you recall, I [suggested](https://github.com/squid-cache/squid/pull/826#pullrequestreview-660368815) changes to the PR description (i.e. the future commit message). It looks like you have not changed it. Do you want me to edit the description to reflect (my understanding of) your [response](https://github.com/squid-cache/squid/pull/826#issuecomment-841730168)? Or perhaps this review request was just for the existing code changes, and you plan to work on the description later?

Sorry, I had not interpreted your commment as advice to change the description. I think you went ahead, thanks for doing that.

> > I can also go after UNUSED_CODE and UNREACHABLECODE.
> 
> Thank you. How about `LINGERING_CLOSE` that is marked (in comm.cc) as "broken, unused, and untested since [933dd09](https://github.com/squid-cache/squid/commit/933dd095fa4602c53703c8bc384c82ed16b86881)"? I know that code wasted quite a bit of time in relatively recent developments. IIRC, it cannot even compile since that ten-year old commit.
> 
> I would also look at `#if SIZEOF_INT64_T == 4` for (hopefully) obvious reasons. If that can be removed, that would remove the whole clientHttpRequestStatus() function and quite a bit of calling code, right?

Sure, I'll take on these as well

Reply:
>> Alex: Please explain what code this PR removes in the PR description.
>> Alex: please ... explicitly mention in the PR description that you left some in on purpose.

> I had not interpreted your commment as advice to change the description.

Sigh. I really do not know how I could have been any more explicit/direct in my change request. What should I say the next time to avoid this kind of misunderstanding?

> I think you went ahead, thanks for doing that.

I did not (you can see PR description editing history by clicking the "edited" link above the description box). Do you want me to? Needless to say, my scope definition may not match how you see your changes (but you can adjust it later, of course).